### PR TITLE
Rustc 1.27 embrace

### PR DESCRIPTION
**Pull request by Pzixel**
This is an attempt to compile `pwasm-std` with nightly `1.27`. I had to implement small fixes in order to achieve it.

### CODE REVIEW

**Comment by fckt** [comment]
great!
Update a `rust-toolchain` file, please

**Comment by Pzixel** [comment]
Any suggestions why demo isn't passing?

File: Cargo.toml [hunk 1]
```diff
@@ -1,6 +1,6 @@
 [package]
 name = "pwasm-std"
-version = "0.6.0"
+version = "0.7.0"
```
Comment (NikVolf) [inline Cargo.toml]: We'd better change this package metadata it in separate pr/commit (i can handle later myself)

File: Cargo.toml [hunk 2]
```diff
@@ -13,9 +13,9 @@ categories = ["no-std", "embedded"]
 exclude = [ "demo/*" ]
 
 [dependencies]
-pwasm-alloc = { path = "alloc", version = "0.3" }
+pwasm-alloc = { path = "alloc", version = "0.4" }
 pwasm-libc = { path = "libc", version = "0.2" }
-tiny-keccak = "1.3"
+tiny-keccak = "1"
```
Comment (NikVolf) [inline Cargo.toml]: seems unrelated to the pr?

File: Cargo.toml (hunk 1 above)
Comment (Pzixel) [reply Cargo.toml]: I think it's ok to change them syncroniosly, because you cannot refer `alloc` package from elsewhere. So this is actually an atomic change in only place. Thus I don't see why it should be separated.

Comment (NikVolf) [reply Cargo.toml]: changes in metadata better keep separated from logic (i mean for all tomls - for alloc and std itself)

File: Cargo.toml (hunk 2 above)
Comment (Pzixel) [reply Cargo.toml]: This overall PR is about updating dependencies and code to make it work with latest compiler. So it's soundly to make this change here for at least two reasons:

1. It's how all other dependencies are written, so it's more consistent with overall library behavior
1. There is no reason to not use latest version if it doesn't break the code
1. Project is currently using version 1.4 (you can check Cargo.lock) so it was misleading and confusing to have here one version one another one is actually is in use.

File: Cargo.toml (hunk 1 above)
Comment (Pzixel) [reply Cargo.toml]: I do believe in the opposite: you can have two packages with same methadata which are completely different, and it's way worse, so I think that changes in logic should always go with different metadata, because otherwise you can break tools that determines if packages are the same by using their metadata. Separating them in different commits breaks tooling expectations.

Comment (NikVolf) [reply Cargo.toml]: Version publish is not bound excusively to your changes, we might get other changes in and publish 0.4 / 0.7 versions all together ;)

Comment (fckt) [reply Cargo.toml]: yeah, I think the practical reason to manage versions in one place becomes clear especially when you have several parallel branches.

**Comment by Pzixel** [comment]
Looking toward publish, because I have to bump version on `pwasm-ethereum` repo as well, and only when it gets updated as well I will be able to finally build my `pwasm-demo0` project :P
